apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+/* global window */
+
+/**
+ * Teacher Application questionaire
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import {Radio, Checkbox, FormControl, FormGroup, ControlLabel} from 'react-bootstrap';
+
+function FieldGroup({ id, label, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+    </FormGroup>
+  );
+}
+
+const grades = ['Kindergarten'].concat(_.range(1,13));
+const subjects = ['Computer Science', 'Computer Literacy', 'Math', 'Science', 'History', 'English', 'Music', 'Art', 'Multimedia', 'Foreign Language'];
+const yesNoResposes = ['Yes', 'No'];
+
+const TeacherApplication = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedCourse: ''
+    };
+  },
+
+  buttonList(type, label, groupName, answers, allowOther = false) {
+    const options = answers.map( (answer, i) => {
+      if (type === 'radio') {
+        return (
+          <Radio
+            value={answer}
+            label={answer}
+            key={i}
+            name={groupName}
+          >
+            {answer}
+          </Radio>
+        );
+      } else if (type === 'check') {
+        return (
+          <Checkbox
+            value={answer}
+            label={answer}
+            key={i}
+            name={groupName}
+          >
+            {answer}
+          </Checkbox>
+        );
+      }
+    });
```

Comment:
Another option since the props passed to `Checkbox` vs `Radio` are the same, you could just do something like:

```jsx
const InputComponent = {radio: Radio, check: Checkbox}[type];
return (
  <InputComponent
    value={answer}
    label={answer}
    key={i}
    name={groupName}
  >
    {answer}
  </InputComponent>
);
```

Or if that seems too obtuse, you could do:

```jsx
const props = {value: answer, label: answer, key: i, name: groupName, children: answer};
return (
  type === 'radio' ? <Radio {...props} /> :
  type === 'check' ? <Checkbox {...props} /> :
  throw "invalid type"
);
```